src/app/clusters/level-control/level-control.cpp

```diff
@@ -123,6 +123,145 @@ static bool shouldExecuteIfOff(EndpointId endpoint, CommandId commandId,
+        uint16_t frequency;
+        VerifyOrReturnError(EMBER_ZCL_STATUS_SUCCESS == Attributes::CurrentFrequency::Get(endpoint, &frequency),
+                            CHIP_ERROR_READ_FAILED);
+
+        AttributeValuePair pairs[scenableAttributeCount];
+
+        // Check for nullable
+        if (!level.IsNull())
+        {
+            pairs[0].attributeID    = Attributes::CurrentLevel::Id;
+            pairs[0].attributeValue = level.Value();
+            pairs[1].attributeID    = Attributes::CurrentFrequency::Id;
+            pairs[1].attributeValue = frequency;
+        }
+        else
+        {
+            pairs[0].attributeID    = Attributes::CurrentFrequency::Id;
+            pairs[0].attributeValue = frequency;
+        }
+
+        app::DataModel::List<AttributeValuePair> attributeValueList(pairs);
+
+        return EncodeAttributeValueList(attributeValueList, serializedBytes);
+    }
+
+    /// @brief Default EFS interaction when applying scene to the OnOff Cluster
+    /// @param endpoint target endpoint
+    /// @param cluster  target cluster
+    /// @param serializedBytes Data from nvm
+    /// @param timeMs transition time in ms
+    /// @return CHIP_NO_ERROR if value as expected, CHIP_ERROR_INVALID_ARGUMENT otherwise
+    CHIP_ERROR ApplyScene(EndpointId endpoint, ClusterId cluster, const ByteSpan & serializedBytes,
+                          scenes::TransitionTimeMs timeMs) override
+    {
+        app::DataModel::DecodableList<Scenes::Structs::AttributeValuePair::DecodableType> attributeValueList;
+
+        ReturnErrorOnFailure(DecodeAttributeValueList(serializedBytes, attributeValueList));
+
+        size_t attributeCount = 0;
+        ReturnErrorOnFailure(attributeValueList.ComputeSize(&attributeCount));
+        VerifyOrReturnError(attributeCount <= scenableAttributeCount, CHIP_ERROR_BUFFER_TOO_SMALL);
+
+        auto pair_iterator = attributeValueList.begin();
+
+        // The level control cluster should have a maximum of 2 attributes
+        uint8_t level = 0;
+        // TODO : Uncomment when frequency is supported by the level control cluster
+        // uint16_t frequency;
+        while (pair_iterator.Next())
+        {
+            auto & decodePair = pair_iterator.GetValue();
+
+            // If attribute ID was encoded, checks which attribute from LC cluster is there
+            switch (decodePair.attributeID)
+            {
+            case Attributes::CurrentLevel::Id:
+                level = static_cast<uint8_t>(decodePair.attributeValue);
+                break;
+            case Attributes::CurrentFrequency::Id:
+                // TODO : Uncomment when frequency is supported by the level control cluster
+                // frequency = static_cast<uint16_t>(decodePair.attributeValue);
+                break;
+            default:
+                return CHIP_ERROR_INVALID_ARGUMENT;
```

Comment:
Here, if you always encode 2 pairs even if you do not initialize 2, I assume attribute id will be default/invalid?